src/remote/remote.go

```diff
@@ -588,12 +588,12 @@ func (c *Client) execute(tid int, target *core.BuildTarget, command *pb.Command,
 	} else if target.IsTextFile {
 		return c.buildTextFile(target, command, digest)
 	}
-	return c.reallyExecute(tid, target, command, digest, needStdout, isTest)
+	return c.reallyExecute(tid, target, command, digest, needStdout, isTest, (c.state.ForceRerun && isTest) || (!isTest && c.state.ForceRebuild))
```

Comment:
Don't think this is quite right - think the build one also needs an `IsOriginalTarget` (c.f. code in `src/build`)